lib/adapters/xhr.js

```diff
@@ -36,13 +40,20 @@ module.exports = function xhrAdapter(resolve, reject, config) {
   request.timeout = config.timeout;
 
   // Listen for ready state
-  request.onload = function handleLoad() {
-    if (!request) {
+  request[loadEvent] = function handleLoad() {
+    if (!request || (request.readyState !== 4 && !xDomain)) {
```

Comment:
Maybe we could extract `(request.readyState !== 4 && !xDomain)` into an explaining variable so it would be easier for people reading the code that understand what this condition is about.

I'm not sure what this does tbh. what does `(request.readyState !== 4 && !xDomain)` mean?